Set up CI with Azure Pipelines [skip ci]

TomAugspurger opened the pull request:
Closes https://github.com/dask/dask-ml/issues/482
Closes https://github.com/dask/dask-ml/issues/484
Closes https://github.com/dask/dask-ml/issues/379

mrocklin commented:
A strong comeback from paternal leave :)  (even if you're only back for 30 minutes or so)

TomAugspurger commented:
Still on leave :)

jrbourbeau commented on azure-pipelines.yml:
```diff
@@ -0,0 +1,63 @@
+  - bash: echo "##vso[task.prependpath]$CONDA/bin"
+    displayName: "Add conda to PATH"
+
+  - bash: conda env create --queit --file=$(envFile) --name=dask-ml-test
```
```suggestion
  - bash: conda env create --quiet --file=$(envFile) --name=dask-ml-test
```

TomAugspurger commented:
This should be about ready.

To summarize

- Moved to Azure pipelines. The primary benefit is testing on windows, which surfaced a few issues with 32-bit systems (LabelEncoder is still unresolved. Skipping for now).
- Removed 2.7 testing (will follow up with proper removal of py2 support).
- We've lost codecov for now. I'll follow up with a PR restoring that.

TomAugspurger commented:
Merging this later today if there's no objections.
